cmd/cluster/core/destroy.go

```diff
@@ -99,7 +100,7 @@ func DestroyCluster(ctx context.Context, hostedCluster *hyperv1.HostedCluster, o
 		if err := c.Patch(ctx, hostedCluster, client.MergeFrom(original)); err != nil {
 			if apierrors.IsNotFound(err) {
 				o.Log.Info("Hosted cluster not found, skipping finalizer update", "namespace", o.Namespace, "name", o.Name)
-			} else {
+			} else if !strings.Contains(err.Error(), "no new finalizers can be added if the object is being deleted") {
```

Comment:
drawback is we make a contract on the error string not being changed. May be do both don't try to add if it's already deleting and also contemplate current PR path?

Reply:
Updated it to also only add the finalizer if the hostedcluster is not deleted it yet, ptal.

The reason this is a string check is that there is no error type for this: https://github.com/kubernetes/kubernetes/blob/d5fdf3135e7c99e5f81e67986ae930f6a2ffb047/staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go#L124